### ISSUE TITLE
Cannot build Reloaded

### ISSUE THREAD

**Issue by ColinPitrat** [cpcsdk/cpctools#72]
Hello,

I tried building Reloaded but I get the following errors:
cpcsdk/cpctools/Reloaded/src/core/../misc/Semaphore.h:29:32: error: ‘sem_delete’ was not declared in this scope

and:
cpcsdk/cpctools/Reloaded/src/core/psg.h:12:22: error: Ym2149Ex.h : No such file or directory

Regards,
Colin

**Comment by pulkomandy** [comment]
For the second one you need to build and install the StSoundLibrary first (it is available in the repo, do not use the official version from Leonard, it is not compatible).
You are right about sem_delete, it should be sem_destroy, not sure what I did here. Will fix this.

**Comment by ColinPitrat** [comment]
OK, building and installing StSoundLibrary and replacing by sem_delete by sem_destroy keeping only the first argument worked. I also had to add #include <libgen.h> to src/snamp/main.cpp so that dirname function was found. Thanks for your quick answer !

In case you don't know it, Travis CI offers a nice and free solution for continuous integration with GitHub: https://travis-ci.org/

You just have to put a .travis.yml in your repo to tell him how to build and it allows to check every change submitted on the repo. In the case of cpcsdk the .travis.yml may need to be a bit complex but for a simple project that I have, it looks like this (mostly listing packages to install and telling how to build):

language: cpp
addons:
 apt:
  packages:
    - libsdl1.2-dev
    - zlib1g-dev
script: 
 - make

Concerning my trial of Reloaded, I now get a core after a windows saying:
"./src/common/sizer.cpp(1401): assert "Assert failure" failed in DoInsert(): too many items (5 > 2*2) in grid sizer (maybe you should omit the number of either rows or columns?)"

I don't know if this rings a bell (if not, don't waste time of it, I'm going to give it a deeper look).

Regards,
Colin

**Comment by pulkomandy** [comment]
No, I didn't hit this one yet, maybe it is from a new version of WxWidgets.
If you want to modify the UI files, they were generated with a patched version of wxFormBuilder (https://github.com/cpcsdk/cpctools/wiki/build%3Areloaded%3Alinux). If you use a version without the patch, the keyboard settings dialog will not work.

I will try to setup a Travis build, it looks like a good idea if we can get it running.

**Comment by pulkomandy** [comment]
I have set up Travis, but it can't build Reloaded because the Travis bots are on Ubuntu 12.04 with a very old version of CMake and we need a newer one.

**Comment by ColinPitrat** [comment]
One possibility is to update cmake. It requires a bit of dirty stuff but some guys are doing it. Here is an example, you may want to try it:
https://github.com/joakimkarlsson/bandit/commit/2f87a5af4a06ff7bb1da90a17e96710db372e03a

**Comment by pulkomandy** [comment]
Yes, we will need a PPA with cmake 3.1.0 for Ubuntu 12.04, and try this.

**Comment by pulkomandy** [comment]
I tried this (with another PPA with a new version of cmake), but it is
not enough, running "cmake" still runs version 2.8.

https://travis-ci.org/cpcsdk/cpctools/builds/101662567

-- 
Adrien.

**Comment by ColinPitrat** [comment]
Right, the PPA I gave you is with cmake 2.8.
You can try this one:
https://launchpad.net/~george-edison55/+archive/ubuntu/precise-backports

**Comment by pulkomandy** [comment]
Yes, this is the one I used. But for some reason, it doesn't work.
